### DISCUSSION TITLE
How to delete data when user requests for user removal and data deletion

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#606] [General Discussion]
Hello all,

Currently using 4.6 SunbirdED.
We've a bunch of users are going to inactive. 
We got a request from client  to remove the usage of those users from sunbird.

**_How to delete data when user requests for user removal and data deletion?_** 

**To remove the user** - We can retire the user using block API
**To remove the content** - Is there any API which can delete the specific content or can delete all the content of a particular channel?

How we can achieve that?
How we can make sure that these contents can be deleted?

cc: @rajeevsathish @reshmi-nair @Krishnaj20

**Comment by amitpriyadarshi** [comment]
@nidhikachhawa24 below are the options:
**To remove the user** - `We can retire the user using block API. `
https://staging.open-sunbird.org/api/user/v1/block
Refer user block api in the [link](https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/apis/user-management)

**To remove the content** - `Is there any API which can delete the specific content or can delete all the content of a particular channel?`
To soft delete the content (status will be set to retired) - refer content retire api in the [link](https://documenter.getpostman.com/view/25463377/2s935hRnae#30e6be1d-6290-4b43-9e77-199f17e0b570)
To soft delete collection (status will be set to retired) - refer collection retire api in the [link](https://documenter.getpostman.com/view/25463377/2s8ZDa32au#2e30059e-8a2b-4063-8a21-288f05c7918a)

To delete all the content of a particular channel, you need to run delete cypher script (to delete contents/collections of particular channel) in neo4j.
Also you need to run cql delete script for all the specific ids in cassandra.

Better approach will be calling retire api for the list of ids.

cc: @Krishnaj20 @vinukumar-vs

**Reply by nidhikachhawa24** [reply]
@amitpriyadarshi Thank you

**Comment by Krishnaj20** [comment]
thanks @amitpriyadarshi 
hope @nidhikachhawa24 your query is answered. Please close this if your query is answered.
